docs/tutorials/cpp.md

```diff
@@ -0,0 +1,92 @@
+# Using ICU4X from C++
+
+_We are still working on improving the user experience of using ICU4X from other languages. As such, this tutorial may be a bit sparse, but we are happy to answer questions on our [discussions forum] and help you out _
+
+ICU4X's core functionality is completely available from C++, with headers generated by [Diplomat].
+
+Generated headers can be found under [`ffi/diplomat/cpp/include`], with Sphinx docs at [`ffi/diplomat/cpp/docs`]. The port is header-only; no additional C++ translation units need to be compiled to use ICU4X from C++.
+
+Typically C++ users can build ICU4X by building the `icu_capi_staticlib` crate ([crates.io][staticlib-crates], [source][staticlib-source]) with the `cpp_default` feature, and link the resultant static library to their C++ application. This crate builds on the `icu_capi` crate: a `no_std` crate containing all of the relevant [Diplomat]-generated `extern "C"` declarations.
+
+Users wishing to use ICU4X on a `no_std` platform will need to write their own crate depending on `icu_capi` that fills in an allocator and panic hooks, similar to what we do in our [freertos port]. ICU4X is happy to accept such ports upstream.
```

Comment:
Thought: We also have the `icu_capi_staticlib/x86tiny` feature. Should we advertise it?